src/hdmf/data_utils.py

```diff
@@ -125,6 +127,272 @@ def maxshape(self):
+            default=None,
+        )
+    )
+    def _get_default_chunk_shape(self, **kwargs):
+        """
+        Select chunk shape with size in MB less than the threshold of chunk_mb.
+
+        Keeps the dimensional ratios of the original data.
+        """
+        chunk_mb = getargs('chunk_mb', kwargs)
+        assert chunk_mb > 0, f"chunk_mb ({chunk_mb}) must be greater than zero!"
+
+        n_dims = len(self.maxshape)
+        itemsize = self.dtype.itemsize
+        chunk_bytes = chunk_mb * 1e6
+        v = np.floor(np.array(self.maxshape) / np.min(self.maxshape))
+        prod_v = np.prod(v)
+        while prod_v * itemsize > chunk_bytes and prod_v != 1:
+            v_ind = v != 1
+            next_v = v[v_ind]
+            v[v_ind] = np.floor(next_v / np.min(next_v))
+            prod_v = np.prod(v)
+        k = np.floor((chunk_bytes / (prod_v * itemsize)) ** (1 / n_dims))
+        return tuple([min(int(x), self.maxshape[dim]) for dim, x in enumerate(k * v)])
+
+    @docval(
+        dict(
+           name="buffer_gb",
+           type=(float, int),
+           doc="Size of the data buffer in gigabytes. Recommended to be as much free RAM as safely available.",
+           default=None,
+        )
+    )
+    def _get_default_buffer_shape(self, **kwargs):
+        """
+        Select buffer shape with size in GB less than the threshold of buffer_gb.
+
+        Keeps the dimensional ratios of the original data.
+        Assumes the chunk_shape has already been set.
+        """
+        buffer_gb = getargs('buffer_gb', kwargs)
+        assert buffer_gb > 0, f"buffer_gb ({buffer_gb}) must be greater than zero!"
+        assert all(np.array(self.chunk_shape) > 0), (
+            f"Some dimensions of chunk_shape ({self.chunk_shape}) are less than zero!"
+        )
+
+        k = np.floor(
+            (buffer_gb * 1e9 / (np.prod(self.chunk_shape) * self.dtype.itemsize)) ** (1 / len(self.chunk_shape))
+        )
+        return tuple([
+            min(max(int(x), self.chunk_shape[j]), self.maxshape[j])
+            for j, x in enumerate(k * np.array(self.chunk_shape))
+        ])
+
+    def recommended_chunk_shape(self) -> tuple:
+        return self.chunk_shape
+
+    def recommended_data_shape(self) -> tuple:
+        return self.maxshape
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        """
+        Retrieve the next DataChunk object from the buffer, refilling the buffer if necessary.
+
+        :returns: DataChunk object with the data and selection of the current buffer.
+        :rtype: DataChunk
+        """
+        if self.display_progress:
+            self.progress_bar.update(n=1)
+        try:
+            buffer_selection = next(self.buffer_selection_generator)
+            return DataChunk(data=self._get_data(selection=buffer_selection), selection=buffer_selection)
+        except StopIteration:
+            if self.display_progress:
```

Comment:
```suggestion
            if self.display_progress:  # pragma: no branch
```
This will make the code coverage tool ignore the fact that the else condition is never encountered.

Reply:
No need according to my local coverage reports; this gets hit by the new tests.